### PR TITLE
Bisect next

### PR DESCRIPTION
**Pull request by pranitbauva1997**
@chriscool @larsxschneider I have successfully solved the problem of exit(). Now there exists some problem with handling different terms and skipped commits. Will get back when I finish that.

### CODE REVIEW

**Comment by pranitbauva1997** [comment]
@chriscool @larsxschneider This functions is causing a lot of problems. I have managed to tackle a lot of them. But now I can't seem to find the reason for failures only. Could you please help??

**Comment by pranitbauva1997** [comment]
@chriscool @larsxschneider I have fixed all the errors. Please review it.

File: builtin/bisect--helper.c [hunk 1]
```diff
@@ -415,57 +538,76 @@ static int bisect_start(struct bisect_terms *terms, int no_checkout,
...
-		} else if (get_oid(argv[i], &oid) && has_double_dash) {
+			die(_("unrecognised option: '%s'"), arg);
+		}
+		else if (get_oid(commit_id, &oid) && has_double_dash) {
 			string_list_clear(&revs, 0);
 			string_list_clear(&states, 0);
-			free(commit_id);
 			die(_("'%s' does not appear to be a valid revision"), argv[i]);
 		} else {
-			free(commit_id);
```
Comment (chriscool) [inline builtin/bisect--helper.c]: Is `commit_id` freed now?

Comment (pranitbauva1997) [inline builtin/bisect--helper.c]: No. It needs to be freed. Will do.

**Pull request closed without merging**